torch/testing/_internal/common_methods_invocations.py

```diff
@@ -12298,8 +12300,24 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            assert_jit_shape_analysis=True,
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
-           sample_inputs_func=sample_inputs_permute,
+           sample_inputs_func=partial(sample_inputs_permute, is_method=False),
            reference_inputs_func=reference_inputs_permute),
+    OpInfo('permute',
```

Comment:
This is an interesting blindspot for our test framework.

The implicit signature for many functions and methods are different, because methods are able to take varargs instead of a single sequence argument. This is not just the case for permute. 

From an OpInfo perspective, we probably don't want to register method variants separately. Method variants are queryable from the "main" OpInfo, after all, and each new OpInfo significantly increases the the testing we do and makes the list of OpInfos less maintainable. Further, I don't think the SampleInputs added here will be used to test the PythonRef for the operation, because it inherits from the "main" OpInfo above.

With infinite time and attention, I think we'd probably write a custom test for methods. We could expand the OpInfos with signatures (possibly take from the corresponding PythonRef?) and use those signatures to determine if the method should have this behavior and, if so, test for it with an explicit test. That could be an issue.

For now, however, I wouldn't add a new OpInfo or expand the SampleInputs, but we could add a test for the correct decomposition behavior between `torch.Tensor.permute` and `_refs.permute` and add it to `test/test_prims.py` with a comment pointing to the bigger issue or simply saying we'd like to generalize the test to handle all such operators.

Reply:
fwiw, we've already dropped attempts at testing varargs for ones/zeros, due to finite time and attention https://github.com/pytorch/pytorch/pull/85117#discussion_r972591968

Reply:
Thank you! I'll add a test and maybe have a quick stab at using the regular opinfo as a basis for the vararg test or so.

Reply:
So I added a varargs test based on @jjsjann123 repro to test_prims.py .
I think one could use the same approach for ones/zeros, but one would have to adapt the call.

What do you think?